python/ray/workers/setup_runtime_env.py

```diff
@@ -78,7 +65,37 @@ def setup(input_args):
                 conda_env_name = get_or_create_conda_env(
                     conda_yaml_path, conda_dir)
 
-        commands += get_conda_activate_commands(conda_env_name)
+        return {"conda_env_name": conda_env_name}
+
+    return {}
+
+
+def setup_worker(input_args):
+    # remaining_args contains the arguments to the original worker command,
+    # minus the python executable, e.g. default_worker.py --node-ip-address=...
+    args, remaining_args = parser.parse_known_args(args=input_args)
+
+    # add worker-shim-pid argument
+    remaining_args.append("--worker-shim-pid={}".format(os.getpid()))
+
+    commands = []
+    runtime_env: dict = json.loads(args.serialized_runtime_env or "{}")
+
+    # Ray client server setups runtime env by itself instead of agent.
+    if runtime_env.get("conda") or runtime_env.get("pip"):
+        if "result" not in runtime_env:
+            result = setup_runtime_env(runtime_env, args.session_dir)
+            runtime_env["result"] = result
+
+    py_executable: str = sys.executable
+
+    result = runtime_env.get("result")
+    if result:
```

Comment:
would the result be empty? If yes, print a warning or something? If not, remove this if